components/local-app/cmd/login.go

```diff
@@ -51,15 +53,19 @@ var loginCmd = &cobra.Command{
 			token = os.Getenv("GITPOD_TOKEN")
 		}
 		if token == "" {
-			var err error
-			token, err = auth.Login(context.Background(), auth.LoginOpts{
-				GitpodURL:   loginOpts.Host,
-				AuthTimeout: 5 * time.Minute,
-				// Request CLI scopes (extended compared to the local companion app)
-				ExtendScopes: true,
-			})
-			if err != nil {
-				return err
+			if !loginOpts.NonInteractive {
```

Comment:
nit: if statements with an else branch should not negate their condition to avoid the double negation when reasoning about the else branch. I.e.